assets/js/components/products-control/index.js

```diff
@@ -2,30 +2,42 @@
  * External dependencies
  */
 import { __, _n, sprintf } from '@wordpress/i18n';
-import { addQueryArgs } from '@wordpress/url';
-import apiFetch from '@wordpress/api-fetch';
 import { Component, Fragment } from '@wordpress/element';
-import { find } from 'lodash';
+import { debounce, find } from 'lodash';
 import PropTypes from 'prop-types';
 import { SearchListControl } from '@woocommerce/components';
 
+/**
+ * Internal dependencies
+ */
+import { isLargeCatalog, getProducts } from './utils';
+
 class ProductsControl extends Component {
 	constructor() {
 		super( ...arguments );
 		this.state = {
 			list: [],
 			loading: true,
 		};
+
+		this.debouncedOnSearch = debounce( this.onSearch.bind( this ), 400 );
```

Comment:
Should the 400 here be a constant somewhere to be more descriptive and so it can be used elsewhere?

Reply:
I think it's fine as a plain number, I think it's pretty clear what's going on.